src/server/context.ts

```diff
@@ -148,7 +148,7 @@ export class ServerContext {
       const url = new URL(req.url);
       if (url.pathname.length > 1 && url.pathname.endsWith("/")) {
         url.pathname = url.pathname.slice(0, -1);
-        return Response.redirect(url, 307);
+        return Response.redirect(JSON.stringify(url), 307);
```

Comment:
```suggestion
        return Response.redirect(url.href, 307);
```

I think this should work too.

Reply:
ok updated 👍